cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90

```diff
@@ -0,0 +1,104 @@
+module shr_lightning_coupling_mod
+
+  !========================================================================
+  ! Module for handling namelist variables related to lightning coupling
+  !========================================================================
+
+  use ESMF         , only : ESMF_VMGetCurrent, ESMF_VM, ESMF_VMGet
+  use ESMF         , only : ESMF_LogFoundError, ESMF_LOGERR_PASSTHRU, ESMF_SUCCESS
+  use shr_sys_mod  , only : shr_sys_abort
+  use shr_log_mod  , only : shr_log_getLogUnit
+  use shr_nl_mod   , only : shr_nl_find_group_name
+  use shr_mpi_mod  , only : shr_mpi_bcast
+
+  implicit none
+  private
+
+  ! !PUBLIC MEMBER FUNCTIONS
+  public shr_lightning_coupling_readnl  ! Read namelist
+
+  character(len=*), parameter :: &
+       u_FILE_u=__FILE__
+
+  !====================================================================================
+CONTAINS
+  !====================================================================================
+
+  subroutine shr_lightning_coupling_readnl(NLFilename, atm_provides_lightning_out)
+
+    !========================================================================
+    ! reads lightning_coupling_nl namelist and returns a variable specifying
+    ! if atmosphere model provides lightning flash frequency field to mediator
+    !========================================================================
+
+    ! input/output variables
+    character(len=*), intent(in)  :: NLFilename ! Namelist filename
+    logical, intent(out) :: atm_provides_lightning_out ! if TRUE atm will provide lightning flash frequency
+
+    !----- local -----
+    logical :: atm_provides_lightning
+    type(ESMF_VM)     :: vm
+    integer           :: unitn                  ! namelist unit number
+    integer           :: ierr                   ! error code
+    logical           :: exists                 ! if file exists or not
+    integer           :: rc
+    integer           :: localpet
+    integer           :: mpicom
+    integer           :: s_logunit
+    character(len=*), parameter :: atm_ozone_frequency_not_present = 'NOT_PRESENT'
+    character(len=*), parameter :: subname = '(shr_lightning_coupling_readnl) '
+    ! ------------------------------------------------------------------
+
+    namelist /lightning_coupling_nl/ atm_provides_lightning
+
+    rc = ESMF_SUCCESS
+
+    !--- Open and read namelist ---
+    if ( len_trim(NLFilename) == 0 ) then
+       call shr_sys_abort( subname//'ERROR: nlfilename not set' )
+    end if
+    call shr_log_getLogUnit(s_logunit)
+    call ESMF_VMGetCurrent(vm, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
```

Comment:
Please change these to use the helper routine
 if (chkerr(rc,__LINE__,u_FILE_u)) return

Reply:
It's in:  use nuopc_shr_methods, only : chkerr

Reply:
fixed